src/scalar.rs

```diff
@@ -19,6 +20,15 @@ impl Default for Scalar {
     }
 }
 
+// `ark_ed_on_bls12_377::Fr` is an alias for `ark_ff::fields::models::Fp256`
+// which has implementations of `From` for many types T. Here we
+// pass those through to `Scalar`.
+impl<T: Into<ark_ed_on_bls12_377::Fr>> From<T> for Scalar {
```

Comment:
General comment for later: we should probably do some thinking about the extent to which we want to expose the arkworks types in the API; it'll probably be unavoidable, but exposing types and traits from your dependencies means you can't be any more stable than they are.